opm/material/fluidmatrixinteractions/EclMaterialLawManager.hpp

```diff
@@ -117,15 +117,60 @@ class EclMaterialLawManager
     {}
 
     void initFromDeck(const Opm::Deck& deck,
-                      const Opm::EclipseState& eclState,
-                      const std::vector<int>& compressedToCartesianElemIdx)
+                      const Opm::EclipseState& eclState)
     {
         // get the number of saturation regions and the number of cells in the deck
         const size_t numSatRegions = eclState.runspec().tabdims().getNumSatTables();
 
+        hasGas = deck.hasKeyword("GAS");
+        hasOil = deck.hasKeyword("OIL");
+        hasWater = deck.hasKeyword("WATER");
```

Comment:
Yeah, i guess I could do that. This serves my purpose (I can call initFromDeck on root process, then I can serialize and broadcast the data), before calling initParamsForElements everywhere. But if you prefer, I can undo the member usage and do that instead.

Reply:
> But if you prefer, I can undo the member usage and do that instead.

I think you should keep the new data members.  It's just their initialisation that I think should be through `EclipseState` rather than through `Deck`.